src/chunker.rs

```diff
@@ -0,0 +1,576 @@
+use futures::prelude::*;
```

Comment:
Here I've moved the existing chunker-related code into its own submodule, to avoid growing a gigantic `lib.rs`. The only change I made to the code was using the futures prelude rather than a bunch of individual imports.